src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/AbstractEntityAIGuard.java

```diff
@@ -1,654 +1,582 @@
+          Items.SHIELD,
+          -1);
+
+        if (shieldSlot != -1
+              && target != null
+              && !target.isDead)
         {
-            targetEntity = this.worker.getLastAttackedEntity();
-            return GUARD_HUNT_DOWN_TARGET;
-        }
-
-        if (targetEntity == null)
-        {
-            targetEntity = BarbarianUtils.getClosestBarbarianToEntity(this.worker, currentSearchDistance);
-        }
+            worker.setHeldItem(EnumHand.OFF_HAND, shieldSlot);
+            worker.setActiveHand(EnumHand.OFF_HAND);
 
-        entityList = CompatibilityUtils.getWorld(worker).getEntitiesWithinAABB(EntityMob.class, this.getTargetableArea(currentSearchDistance));
-        entityList.addAll(CompatibilityUtils.getWorld(worker).getEntitiesWithinAABB(EntitySlime.class, this.getTargetableArea(currentSearchDistance)));
-        entityList.addAll(CompatibilityUtils.getWorld(worker).getEntitiesWithinAABB(EntityPlayer.class, this.getTargetableArea(currentSearchDistance)));
+            worker.faceEntity(target, (float) TURN_AROUND, (float) TURN_AROUND);
+            worker.getLookHelper().setLookPositionWithEntity(target, (float) TURN_AROUND, (float) TURN_AROUND);
 
-        if (targetEntity != null && targetEntity.isEntityAlive() && worker.getEntitySenses().canSee(targetEntity))
-        {
-            return GUARD_HUNT_DOWN_TARGET;
-        }
-
-        setDelay(BASE_DELAY);
-        if (entityList.isEmpty())
-        {
-            if (currentSearchDistance < getMaxVision())
+            if (worker.getDistance(target) > getAttackRange())
             {
-                currentSearchDistance += START_SEARCH_DISTANCE;
-            }
-            else
-            {
-                currentSearchDistance = START_SEARCH_DISTANCE;
-                return GUARD_PATROL;
+                worker.isWorkerAtSiteWithMove(target.getPosition(), getAttackRange());
             }
 
-            return GUARD_SEARCH_TARGET;
         }
-        return GUARD_GET_TARGET;
-    }
 
-    public boolean huntDownlastAttacker()
-    {
-        if (this.worker.getLastAttackedEntity() != null && this.worker.getLastAttackedEntityTime() >= worker.ticksExisted - ATTACK_TIME_BUFFER
-              && this.worker.getLastAttackedEntity().isEntityAlive())
-        {
-            return this.worker.getLastAttackedEntity() != null && this.worker.canEntityBeSeen(this.worker.getLastAttackedEntity());
-        }
-        worker.setLastAttackedEntity(null);
-        return false;
-    }
-
-    private AxisAlignedBB getTargetableArea(final double range)
-    {
-        return this.worker.getEntityBoundingBox().expand(range, HEIGHT_DETECTION_RANGE, range);
+        return GUARD_ATTACK_PHYSICAL;
     }
 
-    /**
-     * Getter for the vision or attack distance.
-     *
-     * @return the max vision.
-     */
-    private double getMaxVision()
+    protected AIState attackPhysical()
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this method to reduce its Cognitive Complexity from 22 to the 20 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)